jOOQ/src/main/java/org/jooq/impl/InsertQueryImpl.java

```diff
@@ -335,18 +371,30 @@ else if (onDuplicateKeyIgnore) {
                 case POSTGRES: {
                     toSQLInsert(ctx);
                     ctx.formatSeparator()
-                       .start(INSERT_ON_DUPLICATE_KEY_UPDATE)
-                       .visit(K_ON_CONFLICT)
-                       .sql(' ');
+                       .start(INSERT_ON_DUPLICATE_KEY_UPDATE);
 
-                    if (onConflict != null && onConflict.size() > 0) {
-                        boolean qualify = ctx.qualify();
+                    if (onConstraint != null) {
+                        ctx.data(DATA_CONSTRAINT_REFERENCE, true);
```

Comment:
Nice one. Curious about this, how did you discover this flag?

Reply:
I've investigated `ConstraintImpl#accept` method. As I see, it's perfectly appropriate in use cases, when it's need to get only `name` of `Constraint`'s object by `Context`

Reply:
Yeah, it's an "acceptable hack" - a workaround if either the rendering API or the data model isn't sophisticated enough to communicate such a "special case" and the special case isn't worth an API change...